python/ray/serve/_private/utils.py

```diff
@@ -428,3 +428,17 @@ def in_interactive_shell():
     import __main__ as main
 
     return not hasattr(main, "__file__")
+
+
+def deprecated(*args, **kwargs):
```

Comment:
can we call this `deprecation_warning` instead of `deprecated` so it's less confusing with the existing ray core-level `Deprecated` decorator?

Reply:
changed it to `guarded_deprecation_warning` for extra verbosity

Reply:
PTAL